upgrades/upgrade_testing_framework/steps/step_report_results.py

```diff
@@ -0,0 +1,115 @@
+import xml.etree.ElementTree as ET
+from typing import Dict, List
+from pathlib import Path
+from itertools import groupby
+import textwrap
+
+from upgrade_testing_framework.core.framework_step import FrameworkStep
+from upgrade_testing_framework.core.expectation import load_knowledge_base, Expectation
+
+
+class ReportResults(FrameworkStep):
+    """
+    This step reports on the results of each expectation.
+    """
+
+    def _run(self):
+        # Get the state we need
+        eligible_expectation_ids = self.state.eligible_expectations
+        test_output_directory = Path(self.state.get_key('test_results_directory'))
+
+        # Begin the step body
+        path_to_knowledge_base = Path("../knowledge_base")
+        knowledge_base = load_knowledge_base(path_to_knowledge_base)
+        expectations: Dict[str, Expectation] = {e.id: e for e in knowledge_base if e.id in eligible_expectation_ids}
+        test_results: Dict[str, Dict] = parse_robot_results(test_output_directory, eligible_expectation_ids)
+
+        # These three lists will collect tuples of (expectation_id, test_phase). If the expectation was skipped
+        # in all phases, test_phase is be None.
+        passed_expectations: List[tuple] = []
+        skipped_expectations: List[tuple] = []
+        failed_expectations: List[tuple] = []
+
+        for expectation_id, results_by_test_phase in test_results.items():
+            if len(results_by_test_phase) == 0:
+                # No results for any stage were found, this expectation likely doesn't have any tests
+                skipped_expectations.append((expectation_id, None))
```

Comment:
It feels like this should be considered an error, as every expectation that applies should have a test result.

Reply:
It seems like it could be the case here where someone has created an expectation but hasn't created the test case yet(or doesn't feel comfortable to). In which case it may still be useful to list to users but not run anything. Maybe there is a stipulation we need that identifies a skipped expectation as one not yet implemented for test cases

Reply:
I'll take a closer look at this and play around with the options.